### PR TITLE
fixes to poa events and added events notifier

### PR DESCRIPTION
**Pull request by BatiGencho**
This PR aims to fix some of the poa-related events and event-streamers which were incorrectly imported from node and additionally introduces a notification client which currently supports sending messages to a slack webhook that would appear in our slack in a "testnet_notifications" channel e.g. The underlying client could also be changed easily.

### CODE REVIEW

File: bin/reth/src/args/rpc_server_args.rs [hunk 1]
```diff
@@ -194,6 +194,12 @@ pub struct RpcServerArgs {
     /// The metrics will be served at the given interface and port.
     #[arg(long, value_name = "BITCOIN_BLOCK_SOURCE", value_parser = parse_url, help_heading = "Btc_block_source")]
     pub btc_block_source: Url,
+
+    /// Notifications Webhook Url
+    ///
+    /// Notifications webhook to send various reports to
+    #[arg(long, value_name = "NOTIFICATIONS_WEBHOOK_URL", value_parser = parse_url, help_heading = "Notifications Webhook url", required = false)]
+    pub notifications_webhook_url: Option<Url>,
```
Comment (arminsabouri) [inline bin/reth/src/args/rpc_server_args.rs]: Nit: consider changing this to be specific for slack
`slack_notifications_webhook_url `

**Review by arminsabouri — COMMENTED** [review]
Nice. Awesome integration. Had one small NIT

**Review by scottmillner — APPROVED** [review]
cool stuff! this could definitely come in handy
🔥 💪

File: bin/reth/src/args/rpc_server_args.rs (hunk 1 above)
Comment (BatiGencho) [reply bin/reth/src/args/rpc_server_args.rs]: fixed

**Review by arminsabouri — APPROVED** [review]

**Pull request merged**